### PR TITLE
[SPARK-30878][SQL][DOC] Improve the CREATE TABLE document

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

Improve the CREATE TABLE document:
1. mention that some clauses can come in as any order.
2. refine the description for some parameters.
3. mention how data source table interacts with data source
4. make the examples consistent between data source and hive serde tables.

### Why are the changes needed?

improve doc

### Does this PR introduce any user-facing change?

no

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @HyukjinKwon @dongjoon-hyun

File: docs/sql-ref-syntax-ddl-create-table-hiveformat.md [hunk 1]
```diff
@@ -96,21 +97,37 @@ CREATE [ EXTERNAL ] TABLE [ IF NOT EXISTS ] table_identifier
 ### Examples
 {% highlight sql %}
 
---Using Comment and loading data from another table into the created table
-CREATE TABLE StudentInfo
```
Comment (cloud-fan) [inline docs/sql-ref-syntax-ddl-create-table-hiveformat.md]: the previous example was wrong. By default `CREATE TABLE` without USING created data source table in 3.0.

Comment (maropu) [reply docs/sql-ref-syntax-ddl-create-table-hiveformat.md]: I think this is an important change for users, so how about describing the difference in 2.4/3.0 here, too?

File: docs/sql-ref-syntax-ddl-create-table.md [hunk 2]
```diff
@@ -20,10 +20,10 @@ license: |
 ---
 
 ### Description
-`CREATE TABLE` statement is used to define a table in an exsisting database. 
+`CREATE TABLE` statement is used to define a table in an existing database. 
 
 The CREATE statements:
-* [CREATE TABLE USING DATASOURCE](sql-ref-syntax-ddl-create-table-datasource.html)
+* [CREATE TABLE USING DATA_SOURCE](sql-ref-syntax-ddl-create-table-datasource.html)
```
Comment (maropu) [inline docs/sql-ref-syntax-ddl-create-table.md]: nit: `DATA_SOURCE` is preferred against `DATASOURCE`? It seems the current codebase has both formats; https://gist.github.com/maropu/a7f780208ff50c990a52121378531c25

**Review by maropu — APPROVED** [review]
LGTM except for the two minor comments.

File: docs/sql-ref-syntax-ddl-create-table.md (hunk 2 above)
Comment (HyukjinKwon) [reply docs/sql-ref-syntax-ddl-create-table.md]: I am confused time to time if it's Datasource or Data source :-).

**Review by HyukjinKwon — APPROVED** [review]

File: docs/sql-ref-syntax-ddl-create-table-hiveformat.md (hunk 1 above)
Comment (HyukjinKwon) [reply docs/sql-ref-syntax-ddl-create-table-hiveformat.md]: I think migration guide should be responsible for it ..

File: docs/sql-ref-syntax-ddl-create-table.md (hunk 2 above)
Comment (maropu) [reply docs/sql-ref-syntax-ddl-create-table.md]: Yea, if we can, better to use one of them consistently.

Comment (cloud-fan) [reply docs/sql-ref-syntax-ddl-create-table.md]: English-wise, "data" and "source" are two words...

Comment (cloud-fan) [reply docs/sql-ref-syntax-ddl-create-table.md]: Let's go with `DATA_SOURCE` in public documents, we can fix the internal naming later.

File: docs/sql-ref-syntax-ddl-create-table-hiveformat.md (hunk 1 above)
Comment (viirya) [reply docs/sql-ref-syntax-ddl-create-table-hiveformat.md]: I remember we already add a note for this in migration guide.

**Review by viirya — APPROVED** [review]

File: docs/sql-ref-syntax-ddl-create-table-hiveformat.md (hunk 1 above)
Comment (cloud-fan) [reply docs/sql-ref-syntax-ddl-create-table-hiveformat.md]: yes we have. Search `CREATE TABLE` in https://github.com/apache/spark/blob/master/docs/sql-migration-guide.md

**Comment by cloud-fan** [comment]
thanks for review, merging to master/3.0!

File: docs/sql-ref-syntax-ddl-create-table-hiveformat.md (hunk 1 above)
Comment (maropu) [reply docs/sql-ref-syntax-ddl-create-table-hiveformat.md]: yea, ok. thanks

**Pull request closed without merging**